automation_tools/satellite6/capsule.py

```diff
@@ -543,8 +545,9 @@ def setup_capsule_content(
     )
     hammer_content_view_publish(content_view_name, organization_id)
     product_id = run(
-        'hammer --csv subscription list --organization-id="{}" '
-        '--search="name=\\"{}\\"" | cut  -d "," -f 8 | grep -vi "ID"',
+        'hammer --csv subscription list --organization-id="{0}" '
+        '--search="name=\"{1}\"" | awk -F, "NR>1{print \$8}"'
```

Comment:
Both \\" and \" don't work:
- \\" doesn't work at all.
- \" doesn't work if name contains a space.
It's beacause a command get wrapped into bin/bash -l -c "<command>" and thus causing unexpected breakage.

```
>>> run('hammer --csv subscription list --organization-id="{0}" --search="name=\\"{1}\\""'.format(1, 'Employee SKU'))
[root@localhost] run: hammer --csv subscription list --organization-id="1" --search="name=\"Employee SKU\""
[root@localhost] out: SKU\": -c: line 0: unexpected EOF while looking for matching `"'
[root@localhost] out: SKU\": -c: line 1: syntax error: unexpected end of file
[root@localhost] out: 


Fatal error: run() received nonzero return code 1 while executing!

Requested: hammer --csv subscription list --organization-id="1" --search="name=\"Employee SKU\""
Executed: /bin/bash -l -c "hammer --csv subscription list --organization-id=\"1\" --search=\"name=\\"Employee SKU\\"\""
```

```
>>> run('hammer --csv subscription list --organization-id="{0}" --search="name=\"{1}\""'.format(1, 'Employee SKU'))
[root@localhost] run: hammer --csv subscription list --organization-id="1" --search="name="Employee SKU""
[root@localhost] out: Error: too many arguments
[root@localhost] out: 
[root@localhost] out: See: 'hammer subscription list --help'
[root@localhost] out: 


Fatal error: run() received nonzero return code 64 while executing!

Requested: hammer --csv subscription list --organization-id="1" --search="name="Employee SKU""
Executed: /bin/bash -l -c "hammer --csv subscription list --organization-id=\"1\" --search=\"name=\"Employee SKU\"\""
```
I figured it out by switching ```'``` and ```"```